### PR TITLE
Doc changes for EdistDistEpochRecordFilter

### PR DESCRIPTION
**Pull request by fkeceli**
- Added some doxygen doc in EditDistEpochRecordFilter.hpp.
- Worked on more in-code documentation additions/fixes.

### CODE REVIEW

File: src/EditDistEpochRecordFilter.hpp [hunk 1]
```diff
@@ -86,19 +86,60 @@ namespace geopm
...
             const double m_unstable_period_hysteresis;
+            // State variable of the stable period detector state machine.
+            bool m_is_period_detected;
+            // Input to the stable period detector state machine.
             int m_last_period;
+            // Input to the stable period detector state machine.
             int m_period_stable;
+            // Input to the stable period detector state machine.
             int m_period_unstable;
-            bool m_is_period_detected;
```
Comment (cmcantalupo) [inline src/EditDistEpochRecordFilter.hpp:98]: If the order of the members is changed, the initializer list also needs to be reordered.

**Review by cmcantalupo — CHANGES_REQUESTED** [review]

File: src/EditDistEpochRecordFilter.hpp (hunk 1 above)
Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp:98]: Ok I am fixing that and updating the pull request.

Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp:98]: I fixed the issue and compiled to make sure it is not broken.

File: src/EditDistEpochRecordFilter.hpp [hunk 2]
```diff
@@ -86,18 +86,59 @@ namespace geopm
...
+            ///
+            /// STATE MACHINE OPERATION
+            /// -------------------------
+            /// PERIOD_DETECTED state means we are observing a stable period of N and as long as we stay in this
+            /// state we expect to observe an epoch distance of length N plus/minus erroneously added/removed
+            /// calls, which are indicated by the edit distance (score).
+            ///
+            /// The period stability is measured by comparing the period detected with the most current record vs
+            /// detected with the previous. We store the previous in m_last_period. Only periods of length
+            /// >= MIN_DETECTABLE_PERIOD are considered as potentially stable
```
Comment (cmcantalupo) [inline src/EditDistEpochRecordFilter.hpp]: "Only periods of length..."  is repeated a few lines down, maybe drop here

File: src/EditDistEpochRecordFilter.hpp [hunk 3]
```diff
@@ -86,18 +86,59 @@ namespace geopm
...
+            ///    * Only periods >= MIN_DETECTABLE_PERIOD are considered as potentially stable.
+            ///
+            /// The conditions for PERIOD_DETECTED -> NO_PERIOD_DETECTED state transition:
+            ///    * Detected period deviated from N for the last UNSTABLE_PERIOD_HYSTERESIS x N records.
+            ///
+            /// The conditions for reporting a record as epoch marker:
+            ///    * "N plus/minus erroneous records" passed since last reported epoch marker.
             bool epoch_detected();
 
+            // The String Edit Distance algorithm that finds the
```
Comment (cmcantalupo) [inline src/EditDistEpochRecordFilter.hpp]: Use /// to enable doxygen

File: src/EditDistEpochRecordFilter.hpp [hunk 4]
```diff
@@ -86,18 +86,59 @@ namespace geopm
...
+            // patterns are implemented in this object.
             std::shared_ptr<EditDistPeriodicityDetector> m_edpd;
+            // Parameter input for the algorithm.
             const int m_min_stable_period;
+            // Parameter input for the algorithm.
```
Comment (cmcantalupo) [inline src/EditDistEpochRecordFilter.hpp]: /// Parameter input from the constructor

?

File: src/EditDistEpochRecordFilter.hpp (hunk 2 above)
Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp]: Commit 55dcff0 fixes this request.

File: src/EditDistEpochRecordFilter.hpp (hunk 3 above)
Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp]: Commit 55dcff0 fixes this request.

File: src/EditDistEpochRecordFilter.hpp (hunk 4 above)
Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp]: Commit 55dcff0 fixes this request.

File: src/EditDistEpochRecordFilter.hpp [hunk 5]
```diff
@@ -82,22 +82,65 @@ namespace geopm
...
+            ///
+            /// The conditions for reporting a record as epoch marker:
+            ///    * "N plus/minus erroneous records" passed since last reported epoch marker.
             bool epoch_detected();
 
+            /// The String Edit Distance algorithm that finds the
+            /// patterns are implemented in this object.
             std::shared_ptr<EditDistPeriodicityDetector> m_edpd;
-            const int m_min_stable_period;
+            // Parameter for the epoch detection algorithm. See
```
Comment (cmcantalupo) [inline src/EditDistEpochRecordFilter.hpp:121]: Are these intended to be doxygen?  /// required?

**Review by cmcantalupo — CHANGES_REQUESTED** [review]

File: src/EditDistEpochRecordFilter.hpp (hunk 5 above)
Comment (fkeceli) [reply src/EditDistEpochRecordFilter.hpp:121]: Intention is not to have doxygen for the private variables but just to make the code more readable.

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**